### PR TITLE
docs: reorganize API refs

### PR DESCRIPTION
**Pull request by vbarda**
This change reorganizes the API ref docs in the following ways:

* Removed any additional text, all of the information is pulled in from the docstrings
* Expanded coverage for all files
* New pages:
  * `Types`
  * `Channels`
  *  `Constants` (only public ones)
* Style changes:
  * Show object types
  * Show type annotations

Example screenshots:

* Graphs:
image

* Errors
image

### CODE REVIEW

File: docs/docs/reference/errors.md [hunk 1]
```diff
@@ -1,6 +1,3 @@
 # Errors
 
-While you may not want to see them, informative errors help you design better workflows.
```
Comment (hinthornw) [inline docs/docs/reference/errors.md:3]: Worth adding a module-level docstring here?

File: docs/docs/reference/channels.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Channels
+
+::: langgraph.channels.base
```
Comment (hinthornw) [inline docs/docs/reference/channels.md:3]: Maybe worth a Module-level docstring saying this section is really primarily for internal or advanced usage?

cc @nfcampos e.g., do we treat changes to these APIs as semver breaking changes?

File: docs/docs/reference/errors.md (hunk 1 above)
Comment (vbarda) [reply docs/docs/reference/errors.md:3]: maybe, what do you think it should say? to me it feels a bit self-explanatory

**Comment by eyurtsev** [comment]
Is this going to break any existing links?

**Comment by eyurtsev** [comment]
This is the code in langchain right now that links to langgraph api reference: https://github.com/langchain-ai/langchain/pull/26877/files

**Comment by vbarda** [comment]
> This is the code in langchain right now that links to langgraph api reference: https://github.com/langchain-ai/langchain/pull/26877/files

@eyurtsev I don't think so, but worth double-checking once we merge

**Pull request merged**